cache_flat_mutation_reader.hh

```diff
@@ -742,8 +796,29 @@ void cache_flat_mutation_reader::copy_from_cache_to_buffer() {
             _current_tombstone = tomb;
             _read_context.cache()._tracker.on_range_tombstone_read();
         }
-        add_to_buffer(_next_row);
-        move_to_next_entry();
+
+        if (remove_row) {
+            _read_context.cache()._tracker.on_row_compacted();
+
+            partition_snapshot_row_weakref row_ref(_next_row);
+            move_to_next_entry();
```

Comment:
True, but we do not touch range tombstones in this patch and shouldn't remove such rows. Is there any situation when we effect range tombstones on remove row tombstones?

Reply:
The scenario I presented is not related to expiring of range tombstones, it's about maintaining correct reader's view on range tombstone boundaries in case we decide to remove the row, and the next entry has a range tombstone. That range tombstone is assumed by this algorithm to apply to the range [_lower_bound, next row position). If we don't update _lower_bound to after_key(), it will stay at before_key(), so the range to which the tombstone applies will be extended. The _lower_bound is interpreted here, when processing the next row entry:

```
            push_mutation_fragment(mutation_fragment_v2(*_schema, _permit, range_tombstone_change(_lower_bound, tomb)));

```

Reply:
Thanks, I've got what do you mean
we can make lower_bound correct like
```
position_in_partition::less_compare less(*_schema);
if (less(_lower_bound, _next_row.position())) {
    _lower_bound = _next_row.position();
}
```